### PR TITLE
fix: Table onChange did not trigger when current reset to first page

### PR DESCRIPTION
**Pull request by 2724635499**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [X] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link
#33487 

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |   onChange is called repeat when total is 0 or null  |
| 🇨🇳 Chinese |   Table 组件 pagination 不设置 total 的情况下，onChange 死循环调用        |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [X] Doc is updated/provided or not needed
- [X] Demo is updated/provided or not needed
- [X] TypeScript definition is updated/provided or not needed
- [X] Changelog is provided or not needed

### CODE REVIEW

File: components/table/hooks/usePagination.ts [hunk 1]
```diff
@@ -73,6 +73,8 @@ export default function usePagination(
   if (mergedPagination.current! > maxPage) {
     // Prevent a maximum page count of 0
     mergedPagination.current = maxPage || 1;
+    maxPage > 0 &&
+      mergedPagination.onChange?.(mergedPagination.current, mergedPagination?.pageSize!);
```
Comment (afc163) [inline components/table/hooks/usePagination.ts:77]: 应该放到下面 onInternalChange 里去。

Comment (afc163) [inline components/table/hooks/usePagination.ts:77]: onChange 不应该放在 render 阶段，还要再想别的思路。

Comment (2724635499) [reply components/table/hooks/usePagination.ts:77]: 我在想想看

**Comment by huanglianbiao** [comment]
今天一个之前的项目重装依赖也出现了类似的情况，把<Table pagination={{onChange}}>改为<Table onChange={onChange}>解决了。

**Pull request closed without merging**